src/cast_core/include/cast_core/actors/AssertiveActor.hpp

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HEADER_9F01CA6C_0D34_4966_A1E0_1BAFCAB49528_INCLUDED
+#define HEADER_9F01CA6C_0D34_4966_A1E0_1BAFCAB49528_INCLUDED
+
+#include <string_view>
+
+#include <mongocxx/pool.hpp>
+
+#include <gennylib/Actor.hpp>
+#include <gennylib/PhaseLoop.hpp>
+#include <gennylib/context.hpp>
+
+#include <metrics/metrics.hpp>
+
+namespace genny::actor {
+
+/**
+ * This actor asserts that the results of two aggregations (or any two valid database commands)
+ * return equivalent results. This is primarily useful for validating the TPC-H workload queries.
+ *
+ * ```yaml
+ * SchemaVersion: 2017-07-01
+ * Actors:
+ * - Name: AssertiveActor
+ *   Type: AssertiveActor
+ *   Phases:
+ *   - Repeat: 1
+ *     Database: test
+ *     Message: coll1 documents with foo=1 are identical to coll2 documents
```

Comment:
(nitpick) The `prefix` creation below adds the word "Assert" here, but it seems non-obvious that would happen, and it's not obvious whether the commands will be logged or not.

Would it make sense to copy the RunCommand actor's [logging configs](https://github.com/mongodb/genny/blob/30fff93566de289bc83e828cdc3f37c2ceaaefb6/src/cast_core/src/RunCommand.cpp#L90)?

And maybe have a separate `OperationMessage` field that prints user-configured messages too.

Reply:
Ah yeah I was just generally linking the top of the config code block for that actor, not the metrics name. Saving more configurability for when it's needed seems reasonable!

If you find the current system useful that seems fine! Was mostly thinking someone might put in a message like `Assert x` and it end up logging as `Assert Assert x`, which wouldn't be the end of the world at all.

Another option could be to make the message something like:

```
AssertiveActor:
    Description: x
    Expected: y
    Actual: z
```

Takes up a bit more screen real estate with newlines and all, but is a bit more explicit, which is useful for failed assertions. Either way totally up to you!

Reply:
Logging on multiple lines seems a bit more difficult to work with when dealing with logkeeper (it helps to keep logs on one line for filtering capabilities). The current format lets you filter for "Assert 1" for e.g. and just see the trace of where exactly the document failed. For now, let's stick to the current version.